File: eva-core/eva-core-log/src/main/java/io/nerv/core/bizlog/supporter/console/ConsoleSupporter.java
package io.nerv.core.bizlog.supporter.console;

import io.nerv.core.bizlog.base.BizLogEntity;
import io.nerv.core.bizlog.base.BizLogSupporter;
import io.nerv.core.bizlog.condition.DefaultSupporterCondition;
import lombok.extern.slf4j.Slf4j;
import org.springframework.context.annotation.Conditional;
import org.springframework.stereotype.Component;

import java.util.Date;
import java.util.List;

/**
 * 控制台日志实现类
 * @author: S.PKAQ
 */
@Slf4j
@Component
@Conditional(DefaultSupporterCondition.class)
public class ConsoleSupporter implements BizLogSupporter {
    private BizLogEntity bizLogEntity;

    public ConsoleSupporter() {
        super();
    }

    public ConsoleSupporter(BizLogEntity  bizLogEntity) {
        this.bizLogEntity = bizLogEntity;
    }

    @Override
    public void save(BizLogEntity bizLogEntity) {

    }

    @Override
    public List<BizLogEntity> getLog() {
        return null;
    }

    @Override
    public List<BizLogEntity> getLogByType(String type) {
        return null;
    }

    @Override
    public List<BizLogEntity> getLogAfter(Date dateTime) {
        return null;
    }

    @Override
    public List<BizLogEntity> getLogBetween(Date begin, Date end) {
        return null;
    }

    @Override
    public void cleanAll() {

    }

    @Override
    public void cleanBefore(Date dateTime) {

    }

    @Override
    public void cleanBetween(Date begin, Date end) {

    }

    @Override
    public void print() {
        log.info("Biz log: " + this.bizLogEntity.getDescription());
    }
}

File: eva-core/src/main/kotlin/io/nerv/core/bizlog/supporter/console/ConsoleSupporter.kt
package io.nerv.core.bizlog.supporter.console

import io.nerv.core.bizlog.base.BizLogEntity
import io.nerv.core.bizlog.base.BizLogSupporter
import io.nerv.core.bizlog.condition.DefaultSupporterCondition
import lombok.extern.slf4j.Slf4j
import org.springframework.context.annotation.Conditional
import org.springframework.stereotype.Component
import java.util.*

/**
 * 控制台日志实现类
 * @author: S.PKAQ
 * @Datetime: 2018/9/26 21:48
 */
@Slf4j
@Component
@Conditional(DefaultSupporterCondition::class)
class ConsoleSupporter : BizLogSupporter {
    private var bizLogEntity: BizLogEntity? = null

    constructor() : super() {}
    constructor(bizLogEntity: BizLogEntity?) {
        this.bizLogEntity = bizLogEntity
    }

    override fun save(bizLogEntity: BizLogEntity?) {}
    override val log: List<BizLogEntity?>?
        get() = null

    override fun getLogByType(type: String?): List<BizLogEntity?>? {
        return null
    }

    override fun getLogAfter(dateTime: Date?): List<BizLogEntity?>? {
        return null
    }

    override fun getLogBetween(begin: Date?, end: Date?): List<BizLogEntity?>? {
        return null
    }

    override fun cleanAll() {}
    override fun cleanBefore(dateTime: Date?) {}
    override fun cleanBetween(begin: Date?, end: Date?) {}
    override fun print() {
        ConsoleSupporter.log.info("Biz log: " + bizLogEntity.getDescription())
    }
}